File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentImpaling.java
package thedarkcolour.futuremc.enchantment;

public class EnchantmentImpaling extends EnchantmentTridentBase {
    public EnchantmentImpaling() {
        super(Rarity.RARE, "impaling");
    }

    @Override
    public int getMinEnchantability(int enchantmentLevel) {
        return 1 + (enchantmentLevel - 1) * 8;
    }

    @Override
    public int getMaxEnchantability(int enchantmentLevel) {
        return getMinEnchantability(enchantmentLevel) + 20;
    }

    @Override
    public int getMaxLevel() {
        return 5;
    }
}

File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentImpaling.kt
package thedarkcolour.futuremc.enchantment

class EnchantmentImpaling : EnchantmentTridentBase(Rarity.RARE, "impaling") {
    override fun getMinEnchantability(enchantmentLevel: Int): Int {
        return 1 + (enchantmentLevel - 1) * 8
    }

    override fun getMaxEnchantability(enchantmentLevel: Int): Int {
        return getMinEnchantability(enchantmentLevel) + 20
    }

    override fun getMaxLevel(): Int {
        return 5
    }
}